pkg/rehearse/jobs.go

```diff
@@ -65,38 +72,45 @@ func makeRehearsalPresubmit(source *prowconfig.Presubmit, repo string, prNumber
 	}
 	rehearsal.Labels[rehearseLabel] = strconv.Itoa(prNumber)
 
-	if len(source.Spec.Containers) != 1 {
-		return nil, fmt.Errorf("cannot rehearse jobs with more than 1 container in Spec")
+	branch := strings.TrimPrefix(strings.TrimSuffix(source.Branches[0], "$"), "^")
+	if source.Agent == KubeAgent {
+		gitrefArg := fmt.Sprintf("--git-ref=%s@%s", repo, branch)
+		rehearsal.Spec.Containers[0].Args = append(source.Spec.Containers[0].Args, gitrefArg)
 	}
-	container := source.Spec.Containers[0]
+	return &rehearsal, nil
+}
 
-	if len(container.Command) != 1 || container.Command[0] != "ci-operator" {
-		return nil, fmt.Errorf("cannot rehearse jobs that have Command different from simple 'ci-operator'")
-	}
+func validateJob(source *prowconfig.Presubmit) error {
```

Comment:
This should be imported from upstream

Reply:
Err, this is not really validation but filtering -- if you want to validate that it's a good job, import. Rename the other checks to be filtersz

Reply:
well, the validation is already happening when we load the prowconfig. So renaming makes sense.